### PR TITLE
Bulk preview

### LINKED ISSUE #201: Bulk preview and publish from Sharepoint and Google drive folders
It would be great if an author could simply select multiple files and documents in SharePoint or Google Drive and bulk preview/publish them.

### PR DESCRIPTION
**Pull request by rofe**
Fix #201

### CODE REVIEW

File: src/extension/lib/process-queue.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+/**
+ * Processes the given queue concurrently. The handler functions can add more items to the queue
+ * if needed.
+ *
+ * @param {Iterable|Array} queue A list of tasks
+ * @param {ProcessQueueHandler} fn A handler function `fn(task:any, queue:array, results:array)`
+ * @param {number} [maxConcurrent = 8] Concurrency level
+ * @returns {Promise<[]>} the results
+ */
+async function processQueue(queue, fn, maxConcurrent = 8) {
```
Comment (tripodsan) [inline src/extension/lib/process-queue.js:35]:
```suggestion
export async function processQueue(queue, fn, maxConcurrent = 8) {
```

File: src/extension/lib/process-queue.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    }
+    handler(value);
+  }
+  // wait until remaining tasks have completed
+  await Promise.all(running);
+  return results;
+}
+
+// module.exports = processQueue;
+export { processQueue };
```
Comment (tripodsan) [inline src/extension/lib/process-queue.js:82]:
```suggestion
```

Comment (tripodsan) [inline src/extension/lib/process-queue.js:82]:
```suggestion
```

**Review by tripodsan — APPROVED** [review]

File: src/extension/lib/process-queue.js (hunk 2 above)
Comment (rofe) [reply src/extension/lib/process-queue.js:82]: I thought an eventual update from the source will be easier if the only difference is the export statement at the very bottom.

**Pull request merged**